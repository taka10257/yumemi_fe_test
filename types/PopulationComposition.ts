export type PopulationComposition = {
  year: number
  value: number
}
