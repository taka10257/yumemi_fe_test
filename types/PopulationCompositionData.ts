export type PopulationCompositionData = {
  year: number
  value: number
}
