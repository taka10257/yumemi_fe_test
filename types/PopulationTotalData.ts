import { PopulationCompositionData } from './PopulationCompositionData'

export type PopulationTotalData = {
  prefCode: number
  prefName: string
  data: PopulationCompositionData[]
}
