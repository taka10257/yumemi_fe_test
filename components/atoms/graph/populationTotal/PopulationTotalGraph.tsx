import React from 'react'
import Highcharts, { SeriesLineOptions } from 'highcharts'
import HighchartsExporting from 'highcharts/modules/exporting'
import HighchartsReact from 'highcharts-react-official'
import { PopulationTotalData } from '../../../../types/PopulationTotalData'

if (typeof Highcharts === 'object') {
  HighchartsExporting(Highcharts)
}

const defaultOptions: Highcharts.Options = {
  title: {
    text: '総人口',
  },
  legend: {
    layout: 'horizontal',
    align: 'center',
    verticalAlign: 'bottom',
  },
  exporting: {
    enabled: false,
  },
  xAxis: {
    title: {
      text: '年度',
      align: 'high',
    },
  },
  yAxis: {
    labels: {
      format: '{value}',
    },
    title: {
      align: 'high',
      offset: 0,
      text: '人口数',
      rotation: 0,
      y: -15,
      x: -15,
    },
  },
}

type PopulationTotalGraphProps = {
  dataList: PopulationTotalData[]
}

export default function PopulationTotalGraph({
  dataList,
}: PopulationTotalGraphProps) {
  return (
    <HighchartsReact
      highcharts={Highcharts}
      options={{ ...defaultOptions, series: createSeries(dataList) }}
    />
  )
}

function createSeries(dataList: PopulationTotalData[]): SeriesLineOptions[] {
  return dataList.map((totalData) => {
    return {
      type: 'line',
      id: totalData.prefCode.toString(),
      name: totalData.prefName,
      data: totalData.data.map((value) => Object.values(value)),
    }
  })
}
