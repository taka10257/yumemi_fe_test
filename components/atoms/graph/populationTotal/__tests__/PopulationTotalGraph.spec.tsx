import React from 'react'
import { PopulationTotalData } from '../../../../../types/PopulationTotalData'
import PopulationTotalGraph from '../PopulationTotalGraph'
import { shallow } from 'enzyme'

describe('components/atoms/graph/populationTotal/PopulationTotalGraph', function () {
  it('表示するデータがないとき', () => {
    const graph = shallow(<PopulationTotalGraph dataList={[]} />)
    expect(graph).toMatchSnapshot()
  })
  it('表示するデータがあるとき', () => {
    const graph = shallow(<PopulationTotalGraph dataList={testData()} />)
    expect(graph).toMatchSnapshot()
  })
})

function testData(): PopulationTotalData[] {
  return [
    {
      prefCode: 1,
      prefName: '北海道',
      data: [
        { year: 1960, value: 5039206 },
        { year: 1965, value: 5171800 },
      ],
    },
    {
      prefCode: 2,
      prefName: '青森',
      data: [
        { year: 1960, value: 1426606 },
        { year: 1965, value: 1416591 },
      ],
    },
  ]
}
