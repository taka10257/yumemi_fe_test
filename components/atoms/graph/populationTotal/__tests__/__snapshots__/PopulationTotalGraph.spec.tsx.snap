// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/atoms/graph/populationTotal/PopulationTotalGraph 表示するデータがあるとき 1`] = `
<Memo(ForwardRef)
  highcharts={[Function]}
  options={
    Object {
      "exporting": Object {
        "enabled": false,
      },
      "legend": Object {
        "align": "center",
        "layout": "horizontal",
        "verticalAlign": "bottom",
      },
      "series": Array [
        Object {
          "data": Array [
            Array [
              1960,
              5039206,
            ],
            Array [
              1965,
              5171800,
            ],
          ],
          "id": "1",
          "name": "北海道",
          "type": "line",
        },
        Object {
          "data": Array [
            Array [
              1960,
              1426606,
            ],
            Array [
              1965,
              1416591,
            ],
          ],
          "id": "2",
          "name": "青森",
          "type": "line",
        },
      ],
      "title": Object {
        "text": "総人口",
      },
      "xAxis": Object {
        "title": Object {
          "align": "high",
          "text": "年度",
        },
      },
      "yAxis": Object {
        "labels": Object {
          "format": "{value}",
        },
        "title": Object {
          "align": "high",
          "offset": 0,
          "rotation": 0,
          "text": "人口数",
          "x": -15,
          "y": -15,
        },
      },
    }
  }
/>
`;

exports[`components/atoms/graph/populationTotal/PopulationTotalGraph 表示するデータがないとき 1`] = `
<Memo(ForwardRef)
  highcharts={[Function]}
  options={
    Object {
      "exporting": Object {
        "enabled": false,
      },
      "legend": Object {
        "align": "center",
        "layout": "horizontal",
        "verticalAlign": "bottom",
      },
      "series": Array [],
      "title": Object {
        "text": "総人口",
      },
      "xAxis": Object {
        "title": Object {
          "align": "high",
          "text": "年度",
        },
      },
      "yAxis": Object {
        "labels": Object {
          "format": "{value}",
        },
        "title": Object {
          "align": "high",
          "offset": 0,
          "rotation": 0,
          "text": "人口数",
          "x": -15,
          "y": -15,
        },
      },
    }
  }
/>
`;
